pkg/operator/osd/osd.go

```diff
@@ -231,11 +236,16 @@ func (c *Cluster) osdContainer(devices []Device, directories []Directory, select
 		envVars = append(envVars, metadataDeviceEnvVar(selection.MetadataDevice))
 	}
 
+	devMountNeeded := len(devices) > 0 || selection.DeviceFilter != "" || selection.getUseAllDevices() || selection.MetadataDevice != ""
```

Comment:
can you set devMountNeeded to true inside the previous `if` block?

Reply:
Good point, I will make the change and push.